client/components/section-header/style.scss

```diff
@@ -35,11 +35,15 @@
 
 .section-header__label {
 	color: $gray;
+	font-size: 14px;
+}
+
+.section-header.header-is-compact .section-header__label {
```

Comment:
It looks like you removed the prop that created this class in the 3rd commit, so let's go ahead and and drop this CSS rule too.